options/posix/generic/netdb-stubs.cpp

```diff
@@ -83,16 +83,23 @@ int getaddrinfo(const char *__restrict node, const char *__restrict service,
 	int addr_count = 1;
 	frg::string<MemoryAllocator> canon{getAllocator()};
 	if (node) {
-		if ((addr_count = mlibc::lookup_name_hosts(addr_buf, node, canon)) <= 0)
-			addr_count = mlibc::lookup_name_dns(addr_buf, node, canon);
-		else
-			addr_count = 1;
+		if ((addr_count = mlibc::lookup_name_ip(addr_buf, node, family)) <= 0) {
+			if (flags & AI_NUMERICHOST)
+			       addr_count = -EAI_NONAME;
+			else if ((addr_count = mlibc::lookup_name_hosts(addr_buf, node, canon)) <= 0)
+				addr_count = mlibc::lookup_name_dns(addr_buf, node, canon);
+			else
+				addr_count = 1;
```

Comment:
This `else` looks a bit strange (but it was also present before). If `lookup_name_hosts` succeeds, we discard the `addr_count` and use 1 instead?

Reply:
This also puzzled me, I could check the source for ``mlibc::lookup_name_hosts()`` again and see why I did that in the first place.

Reply:
Okay, after reading the code again I _think_ it's because there can be multiple entries in ``/etc/hosts`` but we only want to take the first one. Although I'm really not sure about it.